### ISSUE TITLE
Question: Is there a way to retrieve corresponding widget for callback methods?

### ISSUE THREAD

**Issue by sangamk** [arguslab/Argus-SAF#48]
For the callbacks ```apk.model.getCallbackMethods```is there a way to find the layout widget that calls that method? If there is one ofcourse.

```
public void click(View view){
}
<button android:onClick="click"/>
```
or for:

```
findById(id).setOnClickListener
```

**Comment by fgwei** [comment]
Yes, the click method should be collected as callback method for getCallbackMethods.

**Comment by sangamk** [comment]
Yes but can I also find out which widget is linked to that method?

**Comment by fgwei** [comment]
I see. You cannot currently do that. But you can achieve that by modifying this file: https://github.com/arguslab/Argus-SAF/blob/68f87fd596a06f5a6b2f6a4303bbc709005df3cd/org.argus.amandroid/src/main/scala/org/argus/amandroid/core/parser/LayoutFileParser.scala

**Comment by sangamk** [comment]
For this -> 
```
public void click(View view){
}
<button android:onClick="click"/>
```
I got it working.

However I can't find out how to find the widget (button2) for these kinds of methods ->

``` Android
        Button b = findViewById(R.id.button2);
        b.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(), ThirdActvity.class));
            }
        })
```
How would I go about doing this? Is there a way to see if onClick method has a parent View and if so which xml corresponds to that view

**Comment by sangamk** [comment]
Anyone?

**Comment by fgwei** [comment]
This onClick method will be automatically collected as a callback method for this component.
To find the view binding you need to do a point-to analysis to figure out when setOnClickListener get called which button object it is called upon and what is the OnClickListener is been registered.

**Comment by sangamk** [comment]
Thx I will give it a try: 
To clarify I am only interested in the type of the widget that calls the method not its name. So android.widget.Button or android.widget.TextView

**Comment by fgwei** [comment]
You may use something like [ExplicitValueFinder](https://github.com/arguslab/Argus-SAF/blob/4fbf518f1cf5e6a4b07c805aa8e4e1cf197ea47a/org.argus.jawa/src/main/scala/org/argus/jawa/alir/util/ExplicitValueFinder.scala)
